src/wazuh_modules/wm_gcp.c

```diff
@@ -312,80 +321,67 @@ void wm_gcp_pubsub_run(const wm_gcp_pubsub *data) {
-                    mtdebug1(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 2:
-                if (line = strstr(line, "- INFO -"), line) {
-                    mtinfo(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 3:
-                if (line = strstr(line, "- WARNING -"), line) {
-                    mtwarn(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 4:
-                if (line = strstr(line, "- ERROR -"), line) {
-                    mterror(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            case 5:
-                if (line = strstr(line, "- CRITICAL -"), line) {
-                    mterror(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
-            default:
-                if (line = strstr(line, "- INFO -"), line) {
-                    mtinfo(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
+    char * parsing_output = output;
+    int debug_level = isDebug();
+
+    for (line = strstr(parsing_output, WM_GCP_LOGGING_TOKEN); line; line = strstr(parsing_output, WM_GCP_LOGGING_TOKEN)) {
+        char * tokenized_line;
+        os_calloc(WM_STRING_MAX, sizeof(char), tokenized_line);
+	char * next_lines;
+
+        line += strlen(WM_GCP_LOGGING_TOKEN);
+        next_lines = strstr(line, WM_GCP_LOGGING_TOKEN);
+
+        int next_lines_chars = next_lines == NULL ? 0 : strlen(next_lines);
+
+        // 1 is added because it's mandatory to consider the null byte
+        int cp_length = 1 + strlen(line) - next_lines_chars > WM_STRING_MAX ? WM_STRING_MAX : 1 + strlen(line) - next_lines_chars;
+        snprintf(tokenized_line, cp_length, "%s", line);
+        if (tokenized_line[cp_length - 2] == '\n') tokenized_line[cp_length - 2] = '\0'; 
+        
+        char *p_line = NULL;
+
+        if (debug_level >= 2) {
+            if ((p_line = strstr(tokenized_line, "- DEBUG - "))) {
+                p_line += 10;
+                mtdebug1(tag, "%s", p_line);
+            }
+        }
+        if (debug_level >= 1) {
+            if ((p_line = strstr(tokenized_line, "- INFO - "))) {
+                p_line += 9;
+                mtinfo(tag, "%s", p_line);
+            }
+        }
+        if (debug_level >= 0) {
+            if ((p_line = strstr(tokenized_line, "- CRITICAL - "))) {
+                p_line += 13;
+                mterror(tag, "%s", p_line);
+            }
+            if ((p_line = strstr(tokenized_line, "- ERROR - "))) {
+                p_line += 10;
+                mterror(tag, "%s", p_line);
+            }
+            if ((p_line = strstr(tokenized_line, "- WARNING - "))) {
+                p_line += 12;
+                mtwarn(tag, "%s", p_line);
+            }
         }
-    }
 
-    os_free(output);
+	parsing_output += cp_length + strlen(WM_GCP_LOGGING_TOKEN) - 1;
```

Comment:
Please, fix line indentation.

Reply:
Fixed in [664a254](https://github.com/wazuh/wazuh/pull/12769/commits/664a2540494692e73d263a588dc3f1ae1e9681c6).